CMakeLists.txt

```diff
@@ -14,7 +14,7 @@ if(NOT (${CMAKE_VERSION} VERSION_LESS "3.13.0"))
 	# CMake 3.13+ has less restrictive rules for target_link_libraries()
 	# Until we make 3.13 as required minimum version we want to
 	# use old behaviour for compatibility
-	cmake_policy(SET CMP0079 OLD)
+	cmake_policy(SET CMP0079 NEW)
```

Comment:
I understand this policy only superficially yet I'm fairly confident that `cmake_minimum_required` should be bumped to 3.13 and that the version conditionals immediately above and below should then be either removed or simplified.

Reply:
Lets make cmake min version 3.13 (IIRC Zephyr also did this a few weeks ago). @cujomalainey would this work for your linking (or do we also change the policy) ?

Reply:
Let me try it out and see, I am fine with that solution as well

Reply:
Fwiw, I notice Zephyr is at ~3.20.x and we should mirror Zephyr going forwards.